tensorboard/webapp/widgets/histogram/histogram_linked_time_fob_test.ts

```diff
@@ -0,0 +1,182 @@
+    it('gets step higher when position is not on a step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepHigher =
+        fixture.componentInstance.getStepHigherThanAxisPosition(1500);
+      expect(stepHigher).toEqual(200);
+    });
+    it('gets step on given position when that position is on a step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepHigher =
+        fixture.componentInstance.getStepHigherThanAxisPosition(3000);
+      expect(stepHigher).toEqual(300);
+    });
+    it('gets highest step when given position is higher than the max step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepHigher =
+        fixture.componentInstance.getStepHigherThanAxisPosition(8000);
+      expect(stepHigher).toEqual(400);
+    });
+    it('gets lower step when given position is lower than the min step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepHigher =
+        fixture.componentInstance.getStepHigherThanAxisPosition(10);
+      expect(stepHigher).toEqual(100);
+    });
+  });
+
+  describe('getStepLowerThanAxisPosition', () => {
+    it('gets step lower when position is not on a step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepLower =
+        fixture.componentInstance.getStepLowerThanAxisPosition(2500);
+      expect(stepLower).toEqual(200);
+    });
+    it('gets step on given position when that position is on a step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepLower =
+        fixture.componentInstance.getStepLowerThanAxisPosition(3000);
+      expect(stepLower).toEqual(300);
+    });
+    it('gets highest step when given position is higher than the max step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepLower =
+        fixture.componentInstance.getStepLowerThanAxisPosition(8000);
+      expect(stepLower).toEqual(400);
+    });
+    it('gets lower step when given position is lower than the min step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepLower =
+        fixture.componentInstance.getStepLowerThanAxisPosition(10);
+      expect(stepLower).toEqual(100);
+    });
+  });
+
+  describe('getAxisPositionFromStep', () => {
+    it('calls the scale function', () => {
+      let fixture = createComponent({});
+      expect(fixture.componentInstance.getAxisPositionFromStep(150)).toBe(1500);
+      expect(temporalScaleSpy).toHaveBeenCalledOnceWith(150);
+    });
+  });
+
+  describe('interaction with base controller', () => {
+    it('properly uses scale when setting fob position', () => {
+      let fixture = createComponent({
+        linkedTime: {start: {step: 300}, end: null},
+      });
+      fixture.detectChanges();
+      let testController = fixture.debugElement.query(
```

Comment:
Considering a new line between 133 and 134. I think it's generally good to separate the "prep" and "expect". It also makes the tests easier to read.